compiler/modules/nback/basic.bal

```diff
@@ -230,34 +230,122 @@ function buildCall(llvm:Builder builder, Scaffold scaffold, bir:CallInsn insn) r
+                                                                    [<llvm:PointerValue>builder.load(scaffold.address(insn.operands[0])),
+                                                                     constInt(scaffold, POINTER_MASK)]);
+    llvm:ConstPointerValue llSignature = scaffold.getFunctionSignatureValue(signature);
+    llvm:Value isExact = buildRuntimeFunctionCall(builder, scaffold, functionIsExactFunction,
+                                                  [llSignature, builder.bitCast(unTaggedPtr, LLVM_FUNCTION_PTR)]);
+    llvm:PointerValue funcStructPtr = builder.bitCast(builder.addrSpaceCast(unTaggedPtr,
+                                                                            LLVM_TAGGED_PTR_WITHOUT_ADDR_SPACE),
+                                                      fnStructPtrTy);
+    llvm:BasicBlock ifExact = scaffold.addBasicBlock();
+    llvm:BasicBlock ifNotExact = scaffold.addBasicBlock();
+    llvm:BasicBlock afterCall = scaffold.addBasicBlock();
+    builder.condBr(isExact, ifExact, ifNotExact);
+    builder.positionAtEnd(ifExact);
+    check buildExactCall(builder, scaffold, insn, afterCall, funcStructPtr, signature);
+    builder.positionAtEnd(ifNotExact);
+    check buildNotExactCall(builder, scaffold, insn, afterCall, funcStructPtr, signature);
+    builder.positionAtEnd(afterCall);
+}
+
+function buildExactCall(llvm:Builder builder, Scaffold scaffold, bir:CallIndirectInsn insn,
+                        llvm:BasicBlock afterCall, llvm:PointerValue funcStructPtr, t:FunctionSignature signature) returns BuildError? {
+    var { returnType, paramTypes } = signature;
+    llvm:Value[] args = check buildFunctionCallArgs(builder, scaffold, paramTypes, paramTypes,
+                                                    from int i in 1 ..< insn.operands.length() select insn.operands[i]);
+    llvm:PointerValue fnGlobalPtr = builder.getElementPtr(funcStructPtr, [constIndex(scaffold, 0),
+                                                                          constIndex(scaffold, 2)],
+                                                          "inbounds");
     llvm:PointerValue funcPtr = <llvm:PointerValue>builder.load(fnGlobalPtr);
     llvm:Value? retValue = buildFunctionCall(builder, scaffold, funcPtr, args);
-    RetRepr retRepr = semTypeRetRepr(signature.returnType);
-    buildStoreRet(builder, scaffold, retRepr, retValue, insn.result);
+    buildStoreRet(builder, scaffold, semTypeRetRepr(returnType), retValue, insn.result);
+    builder.br(afterCall);
 }
 
-function buildFunctionCallArgs(llvm:Builder builder, Scaffold scaffold, t:SemType[] paramTypes, t:SemType[] instantiatedParamTypes, bir:Operand[] args) returns llvm:Value[]|BuildError {
-    return from int i in 0 ..< args.length()
-           select check buildWideRepr(builder, scaffold, args[i], semTypeRepr(paramTypes[i]), instantiatedParamTypes[i]);
+function buildNotExactCall(llvm:Builder builder, Scaffold scaffold, bir:CallIndirectInsn insn,
```

Comment:
`Inexact` or `NonExact` sounds better to me.

Reply:
Fixed